src/Swarm/TUI/Model.hs

```diff
@@ -939,20 +942,28 @@ data AppOpts = AppOpts
 -- | Initialize the 'AppState'.
 initAppState :: AppOpts -> ExceptT Text IO AppState
 initAppState AppOpts {..} = do
-  let skipMenu = isJust userScenario || isJust toRun || isJust userSeed
+  let isRunningInitialProgram = isJust scriptToRun || autoPlay
+      skipMenu = isJust userScenario || isRunningInitialProgram || isJust userSeed
   gs <- initGameState
   ui <- initUIState (not skipMenu) cheatMode
   let rs = initRuntimeState
   case skipMenu of
     False -> return $ AppState gs ui rs
     True -> do
       (scenario, path) <- loadScenario (fromMaybe "classic" userScenario) (gs ^. entityMap)
+
+      let maybeAutoplay = do
+            guard autoPlay
+            soln <- scenario ^. scenarioSolution
+            return $ SuggestedSolution soln
+      let realToRun = maybeAutoplay <|> (ScriptPath <$> scriptToRun)
```

Comment:
```suggestion
      let realToRun = (guard autoPlay >> SuggestedSolution <$> (scenario ^. scenarioSolution)) <|> (ScriptPath <$> scriptToRun)
```

Something like this (possibly split across a couple lines) seems like it would be simpler/clearer to me, but it's just a matter of taste, I don't feel strongly about it.